### PR TITLE
Improvement to existing nouislider type definition.

### PR DESCRIPTION
**Pull request by guustnieuwenhuis**
I've update the type definitions to be in line with the new version (9.0.0) of nouislider.
The changelog of the new version can be found on:
https://github.com/leongersen/noUiSlider/blob/master/README.md

The type definition wasn't complete either, so I've reviewed the whole definition.

I've also included nearly all examples from the nouislider documentation in the test file.

### CODE REVIEW

File: nouislider/nouislider.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for nouislider v8.0.2
+// Type definitions for nouislider v9.0.0
 // Project: https://github.com/leongersen/noUiSlider
-// Definitions by: Patrick Davies <https://github.com/bleuarg>
+// Definitions by: Guust Nieuwenhuis <https://github.com/lagaffe>
```
Comment (yuit) [inline nouislider/nouislider.d.ts]: Could you append your name instead?

File: nouislider/nouislider.d.ts [hunk 2]
```diff
@@ -16,23 +16,23 @@ declare namespace noUiSlider {
...
+         */
         range: Object;
         /**
          * noUiSlider offers several ways to handle user interaction.
          * The range can be set to drag, and handles can move to taps.
          * All these effects are optional, and can be enable by adding their keyword to the behaviour option.
          * This option accepts a "-" separated list of "drag", "tap", "fixed", "snap" or "none".
          */
-        connect?: string | boolean;
+        connect?: boolean | boolean[];
```
Comment (yuit) [inline nouislider/nouislider.d.ts]: I may misunderstand something but should this be just `boolean[]` as in this doc

> Removed: connect: 'lower' and connect: 'upper'. These settings are replaced by connect: [true, false]; 

Also should the comment be updated?

Comment (Lagaffe) [inline nouislider/nouislider.d.ts]: please check https://refreshless.com/nouislider/slider-options/

**Default**
false
**Accepted values**
true,
false,
array[...]

But I'll update the comment

**Comment by Lagaffe** [comment]
@yuit Can you review my corrections?

**Comment by paulvanbrenk** [comment]
@yuit if this looks good, please merge in `master` and `types-2.0`

**Comment by JuusoPalander** [comment]
Hey! Any idea when this will be merged to types-2.0?

**Comment by andy-ms** [comment]
Soon. See #13233.

**Pull request merged**